app/services/TripJoinRequestStageManager.rb

```diff
@@ -1,38 +1,58 @@
+require "./app/services/InvalidStageChange"
+
 class TripJoinRequestStageManager
   REQUESTER = 1
-  def self.accept!(req)
-    if req.requested?
-      trip = req.trip
-      seats = trip.available_seats - (req.companions + REQUESTER)
-      trip.update_column(:available_seats, seats)
+  def self.accept!(trip_request)
+    if trip_request.requested?
+      trip = trip_request.trip
+      new_seats = trip.available_seats - (trip_request.companions + REQUESTER)
+      trip.update_column(:available_seats, new_seats)
+      new_pets = trip.pets - trip_request.pets
+      trip.update_column(:pets, new_pets)
+
+      user_overlapping_requests = trip_request.user.trip_join_requests.by_stage_requested.where("trip_join_requests.id != #{trip_request.id}").by_trip_date(trip.departure_date)
+      user_overlapping_requests.map(&:rejected!)
+
+      all_trip_requests = trip.trip_join_requests.by_stage_requested.where("trip_join_requests.id != #{trip_request.id}")
+      all_trip_requests.map do |each_request|
```

Comment:
good! but move this pieces of code into a separate function like you said: `reject_overlaping_requests`